src/ekka_dist.erl

```diff
@@ -70,8 +79,27 @@ childspecs() ->
     with_module(fun(M) -> M:childspecs() end).
 
 with_module(Fun) ->
-    Proto = application:get_env(ekka, proto_dist, inet_tcp),
-    Fun(list_to_existing_atom(atom_to_list(Proto) ++ "_dist")).
+    try
+        Proto = resolve_proto(),
+        Module = list_to_atom(Proto ++ "_dist"),
+        Fun(Module)
+    catch
+        C:E->
+            %% this exception is caught by net_kernel
+            error({failed_to_call_ekka_dist_module, C, E})
+    end.
+
+resolve_proto() ->
+    Default = atom_to_list(application:get_env(ekka, proto_dist, inet_tcp)),
+    %% the -proto_dist boot arg is 'ekka'
+    %% and there is a lack of a 'ekka_tls' module
+    %% Also when starting remote console etc, there is no application env to
+    %% read from, so we have to find another way to pass the module
+    case os:getenv("EKKA_PROTO_DIST_MOD") of
+        false -> Default;
+        "" -> Default;
+        Mod -> Mod
+    end.
```

Comment:
Better match the result against the expected modules, to avoid passing incorrect setting? Errors produced by such mistake will be hard to troubleshoot.